src/Microsoft.ML.TimeSeries/STL/TrendInfo.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Microsoft.ML.TimeSeries
+{
+    /// <summary>
+    /// the trend component of time series. and corresponding mathematical properties of the trend. A default rank is available when this instance is constructed.
+    /// </summary>
+    public class TrendInfo : SingleSeriesInfo
+    {
+        private const double _slopeThreshold = 0.25;
+        private const double _msrThreshold = 0.1;
+        private const int _shortTimeseriesLength = 40;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TrendInfo"/> class.
+        /// insight item for trend component
+        /// </summary>
+        /// <param name="x">x-axis values of original curve</param>
+        /// <param name="y">y-axis values of original curve</param>
+        /// <param name="trend">the curve of trend</param>
+        /// <param name="mrs">mean residual squares. which measure the quality of regression model fitting to original curve.</param>
+        public TrendInfo(IReadOnlyList<double> x, IReadOnlyList<double> y, IReadOnlyList<double> trend, double mrs)
+        {
+            X = x;
+            Y = y;
```

Comment:
It seems this constructor is not called anywhere. Please see my comment here:
https://github.com/dotnet/machinelearning/pull/5202/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csv#r440494174

Reply:
Removed this file.

---
In reply to: [440514112](https://github.com/dotnet/machinelearning/pull/5202#discussion_r440514112) [](ancestors = 440514112)